tests/test_project.py

```diff
@@ -478,6 +482,93 @@ def test_project_hyphen_named_environment(project_directory_factory):
         )
     )
     assert project.default_environment == project.environments["my-env"]
+    assert (
+        project.environments["my-env"].prefix == project.directory / "envs" / "my-env"
+    )
+
+
+def test_project_environment_env_path_specified(
+    tmp_path, project_directory_factory, monkeypatch
+):
+    test_envs_path = tmp_path / "apples"
+    test_envs_path.mkdir()
+    monkeypatch.setenv("CONDA_PROJECT_ENVS_PATH", str(test_envs_path))
+
+    env_yaml = f"dependencies: []\nplatforms: [{current_platform()}]"
+    project_yaml = dedent(
+        f"""\
+        name: test
+        environments:
+          my-env: [environment{project_directory_factory._suffix}]
+        """
+    )
+    project_path = project_directory_factory(
+        env_yaml=env_yaml, project_yaml=project_yaml
+    )
+    project = CondaProject(project_path)
+
+    assert project.environments["my-env"].prefix == test_envs_path / "my-env"
+
+
+def test_project_environment_env_path_uses_first_writable(
+    tmp_path, project_directory_factory, monkeypatch
+):
+    test_envs_path1 = tmp_path / "apples"
+    test_envs_path1.mkdir(mode=0o555)
```

Comment:
Is this a valid way of creating a read-only file when these tests are running in Windows? The failure now seems to be that the test project is using the "apples" directory

Reply:
actually, no. I forgot that you can't do that in windows through python

https://github.com/Anaconda-Platform/anaconda-project/blob/6bc61a599c7e3a6bade891cbc1553a161d880ee9/anaconda_project/requirements_registry/providers/test/test_conda_env.py#L450

Reply:
here's the fixture from anaconda-project

https://github.com/Anaconda-Platform/anaconda-project/blob/6bc61a599c7e3a6bade891cbc1553a161d880ee9/anaconda_project/requirements_registry/providers/test/test_conda_env.py#L361-L376